Firestore/core/src/api/settings.h

```diff
@@ -131,14 +133,78 @@ class PersistentCacheSettings : public LocalCacheSettings {
   int64_t size_bytes_;
 };
 
+class MemoryGargabeCollectorSettings {
+ public:
+  enum class MemoryGcKind { kEagerGc, kLruGc };
+  virtual ~MemoryGargabeCollectorSettings() = default;
+  friend bool operator==(const MemoryGargabeCollectorSettings& lhs,
+                         const MemoryGargabeCollectorSettings& rhs);
+  virtual size_t Hash() const = 0;
+
+  MemoryGcKind kind() const {
+    return kind_;
+  }
+
+ protected:
+  explicit MemoryGargabeCollectorSettings(MemoryGcKind kind)
+      : kind_(std::move(kind)) {
```

Comment:
IIRC enum values are unsigned integers. There's probably little/no performance improvement for move vs copy for an integer. No need to change it though.

Reply:
Makes sense.